### ISSUE TITLE
ReplicaSet Name is not configurable on UI

### ISSUE THREAD

**Issue by Swissbite** [Studio3T/robomongo#1288]
Create a working configuration for connecting to a replica set on [mlab.com](https://mlab.com) is not possible just with the GUI.

I had to edit the **robomongo.json** file to and add update the **setName** Property within the connection settings.

**Comment by simsekgokhan** [comment]
Hi @Swissbite , first of all "setName" in the config file, is the cached set name and internal usage only for faster connections. And, it needs be called "cachedSetName", we will do this change probably in the next releases to avoid future confusions. 

Normally, it should be possible to connect without set name. For example mongo atlas cloud service is connectable from Robomongo without entering set name.

I will tag this issue as 'critical', since you are unable to connect and do your work.
We will investigate replica set connection to mlab and try to send you a working mlab example.

**Comment by Swissbite** [comment]
Hi @simsekgokhan 
You can test with the public flip-flop replicationSet from mlab: http://flip-flop.mlab.com/

**Comment by simsekgokhan** [comment]
Thanks a lot, that public set made investigation much faster.

First of all, I have reproduced the problem, seems like the problem is Robomongo cannot learn the set name which is a critical information for connections to replica sets. We will investigate this, seems like a major problem at first look at the code. This is the cause of this problem:

```
rs-flip-flop:PRIMARY> rs.status()
{
        "ok" : 0,
        "errmsg" : "not authorized on admin to execute command { replSetGetStatus: 1.0 }",
        "code" : 13
}
```

So, if a user that can execute "rs.status()" command is used, I believe Robomongo will be able to connect.

**Workaround Solution:**
--------------------

Seems like you were on the correct way, the cached set name saves the day.
1. Close Robomongo.
2. Please find your connection in `/1.0/robomongo.json` and configure `setName` property with the correct set name:
`    
"setName" : "rs-flip-flop"
`
3. Open Robomongo

**Note:**
After any edit/save connection settings, failed connections or cloned connection settings, the cached set name will be cleared to empty string. For more details: please see "Cached set name feature" section on our blog: http://blog.robomongo.org/robomongo-1-rc1/

I was able to connect after these steps. 
Please let us know if it will work for you too.

![screenhunter_89 feb 23 16 52](https://cloud.githubusercontent.com/assets/19390749/23261798/10ba7d1e-f9e9-11e6-841c-2563ffdefce3.jpg)

**Comment by mbidewell** [comment]
I am also encountering this issue.  The workaround noted does work for me.

**Comment by Swissbite** [comment]
After setting the field "setName" in the json file, the connection worked and I'm able to do my work.

**Comment by altairzuzu** [comment]
I can't connect to my free Atlas cluster.
Got "Error: Establish connection failed. SSL tunnel failure: Network is unreachable or SSL connection rejected by server"

"setName" isn't helping too.
SSL with self-signed certificate.
I've changed username, password and host name below.

  {
   "connectionName" : "Cluster0",
   "credentials" : [
    {
     "databaseName" : "admin",
     "enabled" : true,
     "mechanism" : "SCRAM-SHA-1",
     "userName" : "xxx",
     "userPassword" : "xxx"
    }
   ],
   "defaultDatabase" : "xxx",
   "isReplicaSet" : true,
   "replicaSet" : {
    "0" : "cluster0-shard-00-00-0xxxx.mongodb.net:27017",
    "1" : "cluster0-shard-00-01-0xxxx.mongodb.net:27017",
    "2" : "cluster0-shard-00-02-0xxxx.mongodb.net:27017",
    "readPreference" : 0,
    "setName" : ""
   },
   "serverHost" : "cluster0-shard-00-00-0xxxx.mongodb.net",
   "serverPort" : 27017,
   "ssl" : {
    "allowInvalidCertificates" : true,
    "allowInvalidHostnames" : false,
    "askPassphrase" : false,
    "caFile" : "",
    "crlFile" : "",
    "pemKeyFile" : "",
    "pemPassPhrase" : "",
    "sslEnabled" : true,
    "useAdvancedOptions" : false,
    "usePemFile" : false
   }
  }

**Comment by bhyde** [comment]
FYI - The robomongo.json file is found in ~/.config/robomongo/1.0/robomongo.json (at least on the Mac).

**Comment by simsekgokhan** [comment]
Hi @altairzuzu , thanks for reporting the issue.
It seems like Mongo Atlas free clusters uses MongoDB 3.4.x. Currently, Robomongo is in preparation to provide support for 3.4. We plan to work on 3.4 support as top priority task not in the following release.
For now, the solution is using other Mongo Atlas plans, they all use MongoDB 3.2.

**Comment by harianus** [comment]
For me explanation was not very clear, so here my addition:

1. Find a way to get your *replica set name*, something like `rs-...`, mlab.com shows it in the URL
1. Close Robomongo
1. Open this file in you editor `~/.config/robomongo/1.0/robomongo.json`
1. Search for the empty `setName` and add the *replica set name* there and save the file
1. Start Robomongo again and it should work

**Comment by simsekgokhan** [comment]
@harianus , thanks a lot for sharing the detailed steps.

**Comment by danafreer** [comment]
Amending "setName" in robomongo.json file doe NOT work for me either. I am using Mongo Atlas 3.4.

**Comment by simsekgokhan** [comment]
Hi @danafreer , please see the comment https://github.com/paralect/robomongo/issues/1288#issuecomment-282634831. It will work for 3.2 or lower versions. Support for 3.4 is planned not for the next release but the following as top priority.

For news about 3.4 support, you can follow us twitter https://twitter.com/Robomongo and/or follow this ticket: https://github.com/paralect/robomongo/issues/1250

**Comment by danafreer** [comment]
Thanks @simsekgokhan - can you say when the release after next will be?

**Comment by simsekgokhan** [comment]
Sure, next release is planned in 1-2 weeks. And the following will depend on the work needed for 3.4 support, I cannot say right now but I might predict 5-6 weeks from now on.

**Comment by heydonovan** [comment]
I've got a client that is using a Windows machine. Where would the `robomongo.json` file be?

**Comment by samhagin** [comment]
The config file can be found at `c:\Users\YourName\.config\robomongo.json`. Reference https://github.com/Studio3T/robomongo/issues/985

**Comment by simsekgokhan** [comment]
@samhagin , fyi, that's the location for very old versions.
@heydonovan , the path is different for different version.

Please see the following wiki page: [Robomongo Config File Guide](https://github.com/Studio3T/robomongo/wiki/Robomongo-Config-File-Guide)
Please also make a copy/backup of your config file just in case before changing, even if it is a very minor change.

**Comment by simsekgokhan** [comment]
Update about this issue, we have already included "Set Name" option into UI for next release 1.0 Final (ETA: 1-2 weeks). Hopefully, there will be no need to edit config file in the new release.

**Comment by danafreer** [comment]
Still 1-2 weeks before release?

**Comment by simsekgokhan** [comment]
Hi @danafreer , sorry for the delay, we have decided to include (and already included) 4-5 fixes and enhancements into next version 1.0 which will have set name configurable on UI. And it is being tested now. ETA is next week. We will probably start coding for 3.4 support next week after release.

**Comment by danafreer** [comment]
I now have 1.0, and still can't connect to Atlas cluster. Is 3.4 supported now? If so, can you provide instructions on how to configure connection? Thanks.

**Comment by simsekgokhan** [comment]
Hi @danafreer , 3.4 is under development and planned to be supported in the next release. 
For updates, please follow this ticket https://github.com/Studio3T/robomongo/issues/1250.

**Comment by simsekgokhan** [comment]
Hi @Swissbite , @harianus , 
Replica set name configurable on UI feature is available on latest release Robomongo 1.0.
Can you confirm if it is working for you?

https://robomongo.org/download
http://blog.robomongo.org/robomongo-1-0/

**Comment by mbidewell** [comment]
Works for me.

**Comment by danafreer** [comment]
Thanks @simsekgokhan, I am using Atlas free cluster as well, so could you summarize how to configure connection please using v1.0?  (can't wait for next release :-)

**Comment by simsekgokhan** [comment]
@danafreer 
I understand that : ) but current version will not connect to 3.4 clusters, it is not related to config. You can use 3.2 but I think it will not be free on Mongo Atlas.

**Comment by simsekgokhan** [comment]
Hi @altairzuzu , @danafreer, please see [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770)

**Comment by Alendorff** [comment]
Just checked Robomongo 1.1 under windows 10 x64 - works fine with Atlas replica set.

Connect url, which provided to you by atlas, looks like:

```
mongodb://<USER>:<PASSWORD>@<MEMBER1>.mongodb.net:27017,<MEMBER2>.mongodb.net:27017,<MEMBER3>.mongodb.net:27017/<DATABASE>?ssl=true&replicaSet=<YOUR-SHARD>&authSource=admin
```

Robomongo GUI:
**Connection** tab
Connection type: `Replica Set`
Set Name: `<YOUR-SHARD>` from connection url
Add your members which separated in url by `,` (<MEMBERx> above)

**Authentication** tab: 
Database: `admin`
Set your username and password
Auth Mechanism: `SCRAM-SHA-1`

In **SSL** tab I specified `Self-signed Certificate`.

With such configuration robomongo 1.1.0-beta connects fine to MongoDB Atlas.
